cmd/dagger/module.go

```diff
@@ -871,13 +970,108 @@ type moduleDef struct {
 	// the ModuleSource definition for the module, needed by some arg types
 	// applying module-specific configs to the arg value.
 	Source *dagger.ModuleSource
+
+	ModRef string
+
+	Dependencies []*moduleDependency
 }
 
+type moduleDependency struct {
+	Name        string
+	Description string
+	Source      *dagger.ModuleSource
+
+	ModRef string
+	RefPin string
+}
```

Comment:
Not really something to deal with now - but it feels frustrating that we have to duplicate all of the structs + graphql queries for all the typedefs and dependency information. Is the reason we don't use our own codegen here because you can't get multiple fields at once?

If so, it feels like we should probably try and rectify that - even if the solution is to have special "cli-codegen" to generate this code. It feels like it's getting a bit complex - and might be worth the refactoring time.

Reply:
Agree on the refactoring, but as I've explained, I'm trying to move fast while things are shifting. This will all be cleaned up.